unicodetools/src/main/java/org/unicode/text/UCD/GenerateConfusables.java

```diff
@@ -2637,9 +2637,19 @@ private static String getReasonFromFilename(String type) {
         return type.substring(dash + 1, period);
     }
 
+    private static Normalizer modNFKD;
+
+    static String getModifiedNFKD(String cf) {
+        if (modNFKD == null) {
+            modNFKD = new Normalizer(UCD_Types.NFKD, Default.ucdVersion());
+            modNFKD.setSpacingSubstitute();
+        }
+        return modNFKD.normalize(cf);
+    }
```

Comment:
optional: A better way to do this in Java is to create a private inner class like
```
    private static final class ModifiedNFKD {
        private static Normalizer INSTANCE;
        private ModifiedNFKD() {
            // INSTANCE = new Normalizer(...) ...
        }

        static String normalize(String s) {
            return ModifiedNFKD.INSTANCE.normalize(s);
        }
    }
```
and then you can use ModifiedNFKD.normalize(s) elsewhere.

Java does thread-safe lazy initialization of an inner class.

(same for the existing code below)

Reply:
Thank you very much for the review. I couldn't use your snippet verbatim since I was getting some null references and couldn't figure out exactly why. I ended up checking for null in normalize().